src/focus.service.ts

```diff
@@ -416,25 +416,35 @@ export class FocusService {
     const { width, height } = rect;
     let { top, left } = rect;
 
-    for (let parent = el.parentElement; parent !== null; parent = parent.parentElement) {
+    for (let parent = el.parentElement; parent !== container.parentElement; parent = parent.parentElement) {
 
       // Special case: treat the body as the viewport as far as scrolling goes.
-      const prect = parent === document.body
-        ? { top: 0, left: 0, height: window.innerHeight, width: window.innerWidth }
+      const prect = parent === container
+        ? {
+          top: Number(window.getComputedStyle(container, null).paddingTop.slice(0, -2)),
+          left: 0,
+          height: container.clientHeight - Number(window.getComputedStyle(container, null).paddingBottom.slice(0, -2)),
```

Comment:
Eh, I guess. You could keep it like this but I would like to have only one call to getComputedStyle (instead of two), also you should be sure to subtract both the top and bottom margin from the `height` and apply the corresponding changes to the `left` and `width` properties